flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/Buffer.java

```diff
@@ -247,41 +247,62 @@
 		/**
 		 * {@link #NONE} indicates that there is no buffer.
 		 */
-		NONE(false, false, false, false),
+		NONE(false, false, false, false, false),
 
 		/**
 		 * {@link #DATA_BUFFER} indicates that this buffer represents a non-event data buffer.
 		 */
-		DATA_BUFFER(true, false, false, false),
+		DATA_BUFFER(true, false, false, false, false),
 
 		/**
 		 * {@link #EVENT_BUFFER} indicates that this buffer represents serialized data of an event.
 		 * Note that this type can be further divided into more fine-grained event types
 		 * like {@link #ALIGNED_CHECKPOINT_BARRIER} and etc.
 		 */
-		EVENT_BUFFER(false, true, false, false),
+		EVENT_BUFFER(false, true, false, false, false),
 
 		/**
 		 * Same as EVENT_BUFFER, but the event has been prioritized (e.g. it skipped buffers).
 		 */
-		PRIORITIZED_EVENT_BUFFER(false, true, false, true),
+		PRIORITIZED_EVENT_BUFFER(false, true, false, true, false),
 
 		/**
 		 * {@link #ALIGNED_CHECKPOINT_BARRIER} indicates that this buffer represents a
 		 * serialized checkpoint barrier of aligned exactly-once checkpoint mode.
 		 */
-		ALIGNED_CHECKPOINT_BARRIER(false, true, true, false);
+		ALIGNED_CHECKPOINT_BARRIER(false, true, true, false, false),
+
+		/**
+		 * {@link #TIMEOUTABLE_ALIGNED_CHECKPOINT_BARRIER} indicates that this buffer represents a
+		 * serialized checkpoint barrier of aligned exactly-once checkpoint mode, that can be time-out'ed
+		 * to an unaligned checkpoint barrier.
+		 */
+		TIMEOUTABLE_ALIGNED_CHECKPOINT_BARRIER(false, true, true, false, true);
```

Comment:
As a set before, I'm not a huge fan of this very specific `DataType`s. (Why don't we have an `DataType` just for `EndOfPartitionEvent`s as well?) I also think that `ALIGNED_CHECKPOINT_BARRIER` is on a different level than all other types.

One option that I could see is to just have `ANNOUNCED_EVENT_BUFFER(false, true, true, false, true)`.